### PR TITLE
Fix some IntelliJ inspections

### PR DESCRIPTION
**Pull request by leventov**
Prepare TeamCity for IntelliJ 2018.3.1 upgrade. Mostly removed redundant exceptions declarations in `throws` clauses.

### CODE REVIEW

File: extensions-contrib/orc-extensions/src/main/java/org/apache/druid/data/input/orc/OrcHadoopInputRowParser.java [hunk 1]
```diff
@@ -246,15 +246,14 @@ static String typeStringFromParseSpec(ParseSpec parseSpec)
     builder.append(parseSpec.getTimestampSpec().getTimestampColumn()).append(":string");
     // the typeString seems positionally dependent, so repeated timestamp column causes incorrect mapping
     if (parseSpec.getDimensionsSpec().getDimensionNames().size() > 0) {
-      builder.append(",");
-      builder.append(String.join(
```
Comment (drcrallen) [inline extensions-contrib/orc-extensions/src/main/java/org/apache/druid/data/input/orc/OrcHadoopInputRowParser.java]: non germane change?

Comment (leventov) [reply extensions-contrib/orc-extensions/src/main/java/org/apache/druid/data/input/orc/OrcHadoopInputRowParser.java]: Should be OK, see Collectors.joining() call below.

Comment (drcrallen) [reply extensions-contrib/orc-extensions/src/main/java/org/apache/druid/data/input/orc/OrcHadoopInputRowParser.java]: It's not about it being ok, it is about mixing in changes that are not germane to the intention of the PR, which is a dangerous process. 

Things like "does this have adequate test coverage" make it harder to review in the context of the pr.

Unless this change is required for the intellij inspections for some reason, can it be moved to its own PR please?

Comment (leventov) [reply extensions-contrib/orc-extensions/src/main/java/org/apache/druid/data/input/orc/OrcHadoopInputRowParser.java]: @drcrallen the very line to which you left your first comment *is* required for IntelliJ inspections. Removing `builder.append(",")` and `builder.append(":string")` before and after this line and instead providing additional arguments to `Collectors.joining()` is not required.

Do you want me to rollback the second part?

Comment (leventov) [reply extensions-contrib/orc-extensions/src/main/java/org/apache/druid/data/input/orc/OrcHadoopInputRowParser.java]: @drcrallen I'll merge this PR next Monday unless you respond.

Comment (drcrallen) [reply extensions-contrib/orc-extensions/src/main/java/org/apache/druid/data/input/orc/OrcHadoopInputRowParser.java]: If it is required for inspections to pass with the new rules then that is the info that I was looking for. It wasn't clear to me how it was required here.

Comment (leventov) [reply extensions-contrib/orc-extensions/src/main/java/org/apache/druid/data/input/orc/OrcHadoopInputRowParser.java]: @drcrallen do you have more comments for the PR?

Comment (drcrallen) [reply extensions-contrib/orc-extensions/src/main/java/org/apache/druid/data/input/orc/OrcHadoopInputRowParser.java]: Nope, never put a blocker on the PR either. I think @clintropolis 's approval is sufficient here

**Pull request merged**